fiddy/derivative_check.py

```diff
@@ -137,3 +142,118 @@ def method(self, *args, **kwargs):
             success=success,
         )
         return derivative_check_result
+
+
+def get_expected_and_test_values(directional_derivatives):
+    expected_values = []
+    test_values = []
+    for direction_index, directional_derivative in enumerate(
+        directional_derivatives
+    ):
+        test_value = directional_derivative.value
+        test_values.append(test_value)
+
+        expected_value = []
+        for output_index in np.ndindex(self.output_indices):
+            element = self.expectation[output_index][direction_index]
+            expected_value.append(element)
+        expected_value = np.array(expected_value).reshape(test_value.shape)
+        expected_values.append(expected_value)
+
+    return expected_values, test_values
+
+
+class HybridDerivativeCheck(DerivativeCheck):
+    """HybridDerivativeCheck.
+
+    The method checks, if gradients are in finite differences range [min, max],
+    using forward, backward and central finite differences for potential
+    multiple stepsizes eps. If true, gradients will be checked for each
+    parameter and assessed whether or not gradients are within acceptable
+    absolute tolerances.
+    """
+
+    method_id = "hybrid"
+
+    def method(self, *args, **kwargs):
+        success = True
+        expected_values, test_values = get_expected_and_test_values(
+            self.derivative.directional_derivatives
+        )
+
+        results_all = []
+        directional_derivative_check_results = []
+        for step_size in range(0, len(expected_values)):
+            approxs_for_param = []
+            grads_for_param = []
```

Comment:
Should this be reset in the inner loop, instead of here in the outer loop?

Reply:
Should be okay as-is.